src/audio/tdfb/tdfb.c

```diff
@@ -116,14 +122,39 @@ static void tdfb_free_delaylines(struct tdfb_comp_data *cd)
 		fir[i].delay = NULL;
 }
 
+static void *tdfb_filter_seek(struct sof_tdfb_config *config, int num_filters)
+{
+	struct sof_fir_coef_data *coef_data;
+	int i;
+	uint8_t *coefp = (uint8_t *)ASSUME_ALIGNED(&config->data[0], 2);
+
+	/* Note: FIR coefficients are int16_t. An uint_8 type pointer coefp
+	 * is used for jumping in the flexible array member structs coef_data.
+	 */
+	for (i = 0; i < num_filters; i++) {
+		coef_data = (struct sof_fir_coef_data *)coefp;
+		coefp += sizeof(*coef_data) + sizeof(int16_t) * coef_data->length;
+	}
+
+	return (void *)coefp;
+}
+
 static int tdfb_init_coef(struct tdfb_comp_data *cd, int source_nch,
 			  int sink_nch)
 {
 	struct sof_fir_coef_data *coef_data;
 	struct sof_tdfb_config *config = cd->config;
+	int16_t *output_channel_mix_beam_off = NULL;
 	int16_t *coefp;
+	uint8_t *dp;
 	int size_sum = 0;
+	int min_delta_idx;
+	int min_delta;
 	int max_ch;
+	int num_filters;
+	int target_az;
```

Comment:
please add comment that units for relevant vars are deg so magic nums are not so magic